### ISSUE TITLE
parserOptions was not passed to context

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#14696]
this is from https://github.com/eslint/eslint/pull/14622#discussion_r647521786

**Tell us about your environment**

Node version: v16.0.0
npm version: v7.10.0
Local ESLint version: v7.28.0 (Currently used)
Global ESLint version: Not found
Operating System: linux 4.19.128-microsoft-standard

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?** default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let ecmaVersion = null;

linter.defineRule("ecma-version", context => ({
    Program() {
        ecmaVersion = context.parserOptions.ecmaVersion;
    }
}));

linter.verify("", {
    rules: {"ecma-version": 2},
  {parserOptions: {ecmaVersion: 5}},
});

assert.strictEqual(ecmaVersion, 5);
```

it seems the ecmaVersion is always `null`.
```bash
AssertionError: expected null to equal 5
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by mdjermanovic** [comment]
Hmm, I had tried all three tests from https://github.com/eslint/eslint/pull/14622#discussion_r647521786 on [aladdin-add:update/ecma-latest](https://github.com/aladdin-add/eslint/tree/update/ecma-latest) and they worked then.

> ```shell
> AssertionError: expected null to equal 5
> ```

This happens when there's a parsing error, so the rule isn't run and `ecmaVersion` keeps its initial value (`let ecmaVersion = null`). Parsing error isn't thrown from `linter.verify()`, it's in the linting result as a lint message item.

**Comment by aladdin-add** [comment]
oh, my bad! I just passed the wrong configs.
